src/schemas/xspec.rnc

```diff
@@ -405,12 +405,14 @@ common-schematron-assert-report-expect-attributes =
     attribute label { text }?,
 
     ## Identify a specific <assert> or <report> using its id attribute or 
-    ## the id attribute of the parent <rule>.
+    ## the id attribute of the ancestor <rule> or <pattern>. The test
+    ## uses the nearest of these id attribute.
     attribute id { xsd:NCName }?,
 
     ## Match a specific role attribute value of an <assert> or <report> or 
-    ## the parent <rule>. Role attribute values are often used to specify 
-    ## 'error', 'fatal', 'warn', 'warning', 'info', 'information'.
+    ## the ancestor <rule> or <pattern>. The test uses the nearest of
```

Comment:
I'm wondering if this `<pattern>` is true.
Setting `pattern/@role` (like `<sch:pattern role="error">`) on Oxygen 26.0 triggers a validation error: `Attribute 'role' is not allowed to appear in element 'sch:pattern'.`
Additionally, I couldn't find any tests related to `pattern/@role`.

On the other hand, setting `pattern/@id` is valid on Oxygen. And it's covered in the tests:
https://github.com/xspec/xspec/blob/2fdb36f85b00b98a77b752062cd3ff1801de7f18/test/schematron-020.xspec#L18-L20

Reply:
Good catch! The `pattern/@role` "feature" can't be used with valid Schematron code, so that feature shouldn't be documented. Your observation that `pattern/@role` triggers an Oxygen validation error also aligns with the definition in https://github.com/Schematron/schema/blob/main/schematron.rnc (in a repo that is linked from schematron.com).

I fixed the verbiage in 0b517aea51b1eccc4a6b8a857d3855711d286ebe and will also edit the PR title to avoid confusion.